src/Widget/Input.php

```diff
@@ -57,7 +58,13 @@ public function config(FormModelInterface $data, string $attribute, array $optio
         $new = clone $this;
         $new->data = $data;
         $new->attribute = $attribute;
-        $new->options = $options;
+        $rules = $data->rules()[$attribute] ?? [];
+        foreach ($rules as $rule) {
+            if ($rule instanceof HtmlOptionsProvider) {
```

Comment:
Aren't rules meant to define validation rules? These would be always `Rule` instances.

Reply:
I don't understand what do you mean.

Reply:
It doesn't work without next PR.